### ISSUE TITLE
Unable to run from Run | Debug buttons

### ISSUE THREAD

**Issue by JeremyDaug** [rust-lang/rust-analyzer#18042]
**rust-analyzer version**: 0.402095-linux-x64/server/rust-analyzer

**rustc version**: 1.50.1

**editor or extension**: VS Code

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

I've been working with Rust-Analyzer for a while but I took a break from my Rust project for about a month. I came back and ran updates, but now it won't run code or tests.

I can run code from the integrated terminal and VSCode directly (F5, cargo run/test), but when trying to run any code from the Run | Debug buttons above Tests or Main, it fails.

On running it gives me the error: "Terminal process failed to launch: Path to shell executable "cargo" does not exist."

Trying to run debug gives me an additional Error:
"Cargo invocation has failed: Error: could not launch cargo: Error: spawn cargo ENOENT."

**Comment by Veykril** [comment]
I don't think it is necessarily related but your rustc version is way too old for that rust-analyzer version so that might be causing issues. Try updating your rust toolchain

**Comment by JeremyDaug** [comment]
Sorry, I mistyped it. It's 1.80.1, not 1.50.

**Comment by Veykril** [comment]
Given the error then it looks like cargo bin directory is not in your home?

**Comment by andrewbrey** [comment]
I have also been experiencing this, and an old issue I found when looking around for others reporting this problem was https://github.com/rust-lang/rust-analyzer/issues/13409. For me, it does appear as though the conditions present in #13409 are the same ones that dictate whether or not I observe the "Run" / "Debug" / "Run tests" working correctly or failing with an error message like: `The terminal process failed to launch: Path to shell executable "cargo" does not exist.`.

If I open a workspace with multiple linked members, then the "codelens button" actions fail with the above error - this include clicking on "Run" from a binary entrypoint as well as "Run tests" from a test module. If I open a workspace that is only a single project and doesn't have linked members, then these "codelens button" actions work as expected.

**Comment by karolisr** [comment]
I had the same issue. The thing that worked for me was disabling "C/C++ Extension Pack" and installing "CodeLLDB".

**Comment by AdrianEdelen** [comment]
same issue here:
`rustc 1.82.0 (f6e511eec 2024-10-15)`
`cargo 1.82.0 (8f40fc59f 2024-08-21)`
arch linux
rust installed via rustup.

run/debug tasks work if I run the task via the command pallet, but the buttons above the main function result in:
`The terminal process failed to launch: Path to shell executable "cargo" does not exist. ` 
and 
`Cargo invocation has failed: Error: could not launch cargo: Error: spawn cargo ENOENT`
for run and debug respectively.

and just for clarification: these are the buttons I am referring to:

![Image](https://github.com/user-attachments/assets/5f4d245c-2af5-4443-88f1-bccda02c9a83)

cargo is in my path and runs from any terminal just fine and is in a normal location:
`❯ which cargo
/home/adrian/.cargo/bin/cargo`

I also added this to settings.json:
```json
"rust-analyzer.cargo.extraEnv": {

        "PATH": "/home/adrian/.cargo/bin:${env:PATH}"
    }
```

**Comment by hgyxbll** [comment]
> same issue here: `rustc 1.82.0 (f6e511eec 2024-10-15)` `cargo 1.82.0 (8f40fc59f 2024-08-21)` arch linux rust installed via rustup.
> 
> run/debug tasks work if I run the task via the command pallet, but the buttons above the main function result in: `The terminal process failed to launch: Path to shell executable "cargo" does not exist. ` and `Cargo invocation has failed: Error: could not launch cargo: Error: spawn cargo ENOENT` for run and debug respectively.
> 
> and just for clarification: these are the buttons I am referring to:
> 
> ![Image](https://github.com/user-attachments/assets/5f4d245c-2af5-4443-88f1-bccda02c9a83)
> 
> cargo is in my path and runs from any terminal just fine and is in a normal location:
> 
> ```
> ❯ which cargo
> /home/adrian/.cargo/bin/cargo
> ```
> 
> I also added this to settings.json:
> 
> "rust-analyzer.cargo.extraEnv": {
> 
>         "PATH": "/home/adrian/.cargo/bin:${env:PATH}"
>     }

I think it may not support ${env:PATH}.
